File: desktop/src/main/java/com/youssef12347/exchange/api/model/Transaction.java
package com.youssef12347.exchange.api.model;

import com.google.gson.annotations.SerializedName;

public class Transaction {
    @SerializedName("usd_amount")
    Float usdAmount;
    @SerializedName("lbp_amount")
    Float lbpAmount;
    @SerializedName("usd_to_lbp")
    Boolean usdToLbp;
    @SerializedName("id")
    Integer id;
    @SerializedName("added_date")
    String addedDate;
    @SerializedName("user_name")
    String userName;
    @SerializedName("receiver_name")
    String receiverName;

    public Transaction(Float usdAmount, Float lbpAmount, Boolean usdToLbp, String receiverName) {
        this.usdAmount = usdAmount;
        this.lbpAmount = lbpAmount;
        this.usdToLbp = usdToLbp;
        this.receiverName = receiverName;
    }

    public Float getUsdAmount() {
        return usdAmount;
    }

    public Float getLbpAmount() {
        return lbpAmount;
    }

    public Boolean getUsdToLbp() {
        return usdToLbp;
    }

    public Integer getId() {
        return id;
    }

    public String getAddedDate() {
        return addedDate;
    }


    public String getUserName() {
        return userName;
    }

    public String getReceiverName() {
        return receiverName;
    }
}

File: android/app/src/main/java/com/kmz07/currencyexchange/api/model/Transaction.kt
package com.kmz07.currencyexchange.api.model

import com.google.gson.annotations.SerializedName
import java.sql.Date
import java.sql.Timestamp
import java.text.DateFormat
import java.util.*

class Transaction() {
    @SerializedName("usd_amount")
    var usdAmount: Float? = null
    @SerializedName("lbp_amount")
    var lbpAmount: Float? = null
    @SerializedName("usd_to_lbp")
    var usdToLbp: Boolean? = null
    @SerializedName("added_date")
    var addedDate: Date? = null
    @SerializedName("id")
    var id: Int? = null
}